src/containers/Tenant/QueryEditor/QueryEditor.js

```diff
@@ -650,7 +660,12 @@ function QueryEditor(props) {
 
 const mapStateToProps = (state) => {
     return {
-        executeQuery: state.executeQuery,
+        executeQuery: {
+            ...state.executeQuery,
+            runAction: state.executeQuery.runAction ||
```

Comment:
I think it will be more clear if `runAction` selection logic would be placed in redux store and here it should be only retrieved.